libpod/container_internal_linux.go

```diff
@@ -1098,6 +1100,160 @@ func (c *Container) addNamespaceContainer(g *generate.Generator, ns LinuxNS, ctr
 	return nil
 }
 
+func (c *Container) addCheckpointImageMetadata(importBuilder *buildah.Builder) error {
+	// Get information about host environment
+	hostInfo, err := c.Runtime().hostInfo()
+	if err != nil {
+		return fmt.Errorf("getting host info: %v", err)
+	}
+
+	criuVersion, err := criu.GetCriuVestion()
+	if err != nil {
+		return fmt.Errorf("getting criu version: %v", err)
+	}
+
+	// Add image annotations with information about the container and the host.
+	// This information is useful to check compatibility before restoring the checkpoint
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationName,
+		c.Config().Name,
```

Comment:
Do not call `c.Config()` just use `c.config`, `Config()` return a deep copy and thus is very slow.